File: GoImplementation/task.go
/*
=========================================
task.go
=========================================
Defines the Task structure used by the data processing system.
Each task has an ID and a payload string. The Process() method simulates work
with a random delay and returns the result string.
*/

package main

import (
	"fmt"
	"math/rand"
	"time"
)

// Task struct represents a unit of work with ID and Payload
type Task struct {
	ID      int
	Payload string
}

// Process simulates processing of a task with a delay
func (t Task) Process() string {
	// Simulate work delay between 500ms to 1500ms
	delay := time.Duration(rand.Intn(1000)+500) * time.Millisecond
	time.Sleep(delay)

	// Return processed result
	return fmt.Sprintf("Processed %s", t.Payload)
}

File: JavaImplementation/src/com/data/processor/Task.java
/*
 * Task.java
 * ---------
 * Represents a single unit of work for the worker threads.
 * Each task has an ID and some payload (simulated here as a string).
 */

 package com.data.processor;

 public class Task {
     private final int taskId;
     private final String payload;
 
     // Constructor
     public Task(int taskId, String payload) {
         this.taskId = taskId;
         this.payload = payload;
     }
 
     // Returns task ID
     public int getTaskId() {
         return taskId;
     }
 
     // Returns task payload
     public String getPayload() {
         return payload;
     }
 
     /*
      * Simulates processing of the task.
      * Includes a delay to mimic computational work.
      */
     public String process() throws InterruptedException {
         // Simulate computation delay
         Thread.sleep((long) (Math.random() * 1000 + 500));
         return "Processed " + payload;
     }
 }